tests/test_operation.py

```diff
@@ -1475,3 +1484,19 @@ def test_composed(self):
         assert not both(self.cnot)
         assert not both(self.rot)
         assert not both(self.exp)
+
+
+class TestStaticProperties:
+    @pytest.mark.parametrize(
+        "op, num_params",
+        [
+            (qml.Identity, 0),
+            (qml.PauliZ, 0),
+            (qml.RY, 1),
+            (qml.TwoModeSqueezing, 2),
+        ],
+    )
+    def test_num_params_can_be_static(self, op, num_params):
+        """Test for a sample of Operation that can have a static num_params
+        property that it is indeed static"""
+        assert op.num_params == num_params
```

Comment:
Wouldn't a much safer test be to create a new operation that defines the class attribute and check that it can be accessed? What you test here is that a few example gates use this mechanism, which is not the same

Reply:
Oh, I just see what I suggest should be covered in the tests above. 

I am not sure what this one adds then.

Reply:
See above, this test case seems to be needed then!

Reply:
It is arguable whether this one is needed. I think there are two extreme points of view: Either remove this test altogether. Or test that all classes that can defined num_params on the class level do so. I was so far too lazy to compile the full. I went for something in between in the hope that this would reduce the chances of this feature being broken again in a future version.